### ISSUE TITLE
Replacing a string in the headline always replaces the find-string with the empty string

### ISSUE THREAD

**Issue by SegundoBob** [leo-editor/leo-editor#2220]
# Overview

I think the find-string can be anything.  I think the replace-string can be anything.  I think the bug always occurs.

# My procedure

Create a new node.
Change the headline to "xxxx yyyy".
Find: "xxxx"
Replace: "2222"
Select any node above the new node.
Press F3 till the "xxxx" in the new node is selected.
Press Ctrl+- to do the replace.
The new headline is " yyyy".

# Test System

Leo Log Window
Leo 6.5-devel, devel branch, build 13f386e942
2021-09-21 15:56:05 -0500
Python 3.8.10, PyQt version 5.12.8
linux

Revision ('devel', '13f386e94213') also contains this bug.

**Comment by SegundoBob** [comment]
The most recent revision with the bug confirmed is:

Leo Log Window
Leo 6.5-devel, devel branch, build c7b039961c
2021-09-23 10:26:53 -0500
Python 3.8.10, PyQt version 5.12.8
linux

**Comment by edreamleo** [comment]
Thanks for this report. I'll fix it immediately.

**Comment by edreamleo** [comment]
According to git bisect, the first bad commit was:
```
commit b46e83ff037cc17654fb5a35337f83e5bde96c57
Author: Edward K. Ream <edreamleo@gmail.com>
Date:   Tue Sep 14 12:21:37 2021 -0500
    Fix #2172
```

**Comment by edreamleo** [comment]
PR #2222 fixes this bug. The PR has been merged into the 6.4 branch, and merged (by hand) into the devel branch.
